crates/engine/tree/src/tree/mod.rs

```diff
@@ -1361,6 +1408,10 @@ where
                 FromOrchestrator::BackfillSyncFinished(ctrl) => {
                     self.on_backfill_sync_finished(ctrl)?;
                 }
+                FromOrchestrator::Terminate { tx } => {
+                    debug!(target: "engine::tree", "received terminate request");
+                    self.pending_termination = Some(tx);
```

Comment:
can we not drive the termination to completion here? this seems easier than yielding control back to the main loop for this?

we can make this function the return a bool if that helps to terminate the task